### PR TITLE
[Processing] second group of vector geometry algotihms

### LINKED ISSUE #1285: [FEATURE][processing] New algorithm for offsetting lines
Original commit: https://github.com/qgis/QGIS/commit/82f4a82c66865e3d0b4af6c9410821c740eb3232 by nyalldawson

Unfortunately this naughty coder did not write a description... :-(

### LINKED ISSUE #1354: [FEATURE][processing] Optimised points along geometry algorithm
Original commit: https://github.com/qgis/QGIS/commit/8db9284cb304618c91c22c380711a0d4dd02f707 by nyalldawson

Supports also polygon geometries, handles null geometries,
and records the original line angle along with the distance
for each point.

### LINKED ISSUE #1355: [FEATURE][processing] Extract nodes algorithm now saves node
index, distance along line and angle at node
Original commit: https://github.com/qgis/QGIS/commit/9fa4e776dbf2ba5004583be1bcb0e8eb8c343301 by nyalldawson

Also correctly handles null geometries

### LINKED ISSUE #1488: [FEATURE][processing] New algorithm to extract specific nodes
Original commit: https://github.com/qgis/QGIS/commit/82312e10e472fb91a5b5c0253456130dea86521e by nyalldawson

This algorithm allows you to extract specific nodes from geometries.
Eg you can extract the first or last node in the geometry.

The algorithm accepts a comma separated list of node indices to
extract, eg 0 = first node, 1 = second node, etc. Negative indices
can be used to extract nodes from the end of the geometry. Eg
-1 = last node, -2 = second last node.

### LINKED ISSUE #1498: [FEATURE] [processing] New algorithm to compute geometry by expression
This algorithm updates existing geometries (or creates new geometries) for input features by use of a QGIS expression. This allows complex geometry modifications which can utilise all the
flexibility of the QGIS expression engine to manipulate and create geometries for output features.

see https://github.com/qgis/QGIS/pull/3692/commits/12ab6c6f6f77aea5cf5e31e9af81723d35a263cb

### LINKED ISSUE #1520: [FEATURE] Add method to calculate pole of inaccessibility for polygons
Original commit: https://github.com/qgis/QGIS/commit/d6f09c012e71cb827c2e4408783f724638094cec by nyalldawson

Implements a method in QgsGeometry and a processing algorithm to
calculate the pole of inaccessibility for a surface, which is the
most distant internal point from the boundary of the surface. This function
uses the 'polylabel' algorithm (Vladimir Agafonkin, 2016), which is an iterative
approach guaranteed to find the true pole of inaccessibility within a specified
tolerance. More precise tolerances require more iterations and will take longer
to calculate.

### LINKED ISSUE #1549: [FEATURE][processing] Remove duplicate create points along lines script
Original commit: https://github.com/qgis/QGIS/commit/6bb4934907ea26bcff802bb4ae9abbb47eae7d5f by nyalldawson

This functionality is duplicated by the PointsAlongGeometry algorithm

Keep the algorithm version, since it has unit tests and supports things
like progress reports.

(marked as feature for inclusion in release notes)

### LINKED ISSUE #1674: [FEATURE][processing] algorithm to fix invalid geometries using native
makeValid() implementation
Original commit: https://github.com/qgis/QGIS/commit/0293bc76a0bb6e7ec55df59b8bb2764204552247 by alexbruy

Unfortunately this naughty coder did not write a description... :-(

### LINKED ISSUE #2069: [FEATURE][Processing] Minimal enclosing circle
Original commit: https://github.com/qgis/QGIS/commit/e30f7044c9af22096dda8722d188d452228d420c by nyalldawson

Unfortunately this naughty coder did not write a description... :-(

### LINKED ISSUE #2076: [FEATURE] New processing algorithm "minimum bounding geometry"
Original commit: https://github.com/qgis/QGIS/commit/83affdc7f531a77cdb963fa8285fdc7af9015c76 by nyalldawson

This algorithm creates geometries which enclose the features
from an input layer.

Numerous enclosing geometry types are supported, including
bounding boxes (envelopes), oriented rectangles, circles and
convex hulls.

Optionally, the features can be grouped by a field. If set,
this causes the output layer to contain one feature per grouped
value with a minimal geometry covering just the features with
matching values.

### PR DESCRIPTION
**Pull request by ghtmtt**
This is the second of 3 PR for the Processing Vector Geometry section. Supersedes #2240 

fix #2069 fix #2076  fix #1674  fix #1498  fix #1488  fix #1355 fix #1549 fix #1354 fix #1520 fix #1285

reviewers are super welcome!

### CODE REVIEW

**Comment by ghtmtt** [comment]
Travis complains (correctly) for a missing anchor. This is due to the splitting of the PR. It should be definitely safe we all 3 PR will be merged

**Comment by ghtmtt** [comment]
Fixes #2069 #2076 #1674 #1498 #1488 #1355

File: source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst [hunk 1]
```diff
@@ -531,421 +531,619 @@ Console usage
...
+lines in the input layer.
 
-Description
-...........
+.. note:: the line layer must have closed shapes in order to be transformed into
+  a polygon
 
-<put algorithm description here>
+.. figure:: /static/user_manual/processing_algs/qgis/polygonize.png
+   :align: center
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst]: how about three separated simple lines (two closed and one unclosed) with yellow (?) polygon in background filling the closed ones?

File: source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst [hunk 2]
```diff
@@ -531,421 +531,619 @@ Console usage
...
+Splits the multipart input layers into single features.
 
-Description
-...........
+The attributes of the output layers are the same of the original ones but divided
+into single features.
 
-<put algorithm description here>
+.. figure:: /static/user_manual/processing_algs/qgis/multipart.png
+   :align: center
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vector_geometry_tools.rst]: I'm not sure we need the attribute table (same for merge vector image). how about simply show the parts in a single color and then in different colors?

**Pull request closed without merging**